Shipr-Containerization/Jenkinsfile

```diff
@@ -0,0 +1,78 @@
+pipeline {
+    agent any
+
+    environment {
+        DOCKER_HUB_REPO = 'osamaazm/test-repo'
+    }
+
+    stages {
+        stage('Branch Selection') {
+            steps {
+                script {
+                    // Print the selected branch
+                    echo "Running pipeline on branch: ${params.BRANCH_NAME}"
+                }
+            }
+        }
+
+        stage('Checkout Code') {
+            steps {
+                script {
+                    // Checkout the code from the selected branch
+                    checkout([
+                        $class: 'GitSCM',
+                        branches: [[name: "*/${params.BRANCH_NAME}"]],
+                    ])
+                }
+            }
+        }
+
+        stage('Pull Images') {
+            steps {
+                script {
+                    // Pull Docker images from Docker Hub
+                    sh '''
+                    docker pull ${DOCKER_HUB_REPO}:shipr-inventory
+                    docker pull ${DOCKER_HUB_REPO}:shipr-payment
+                    docker pull ${DOCKER_HUB_REPO}:shipr-frontend
+                    '''
+                }
+            }
+        }
+
+        stage('Stop Existing Containers') {
+            steps {
+                script {
+                    // Stop and remove any existing containers
+                    sh '''
+                    docker rm -f shipr-inventory || true
+                    docker rm -f shipr-payment || true
+                    docker rm -f shipr-frontend || true
+                    '''
+                }
+            }
+        }
+
+        stage('Run Containers') {
+            steps {
+                script {
+                    // Run the services as Docker containers
+                    sh '''
+                    docker run -d --name shipr-inventory -p 3001:3001 ${DOCKER_HUB_REPO}:shipr-inventory
+                    docker run -d --name shipr-payment -p 3002:3002 ${DOCKER_HUB_REPO}:shipr-payment
+                    docker run -d --name shipr-frontend -p 3000:3000 ${DOCKER_HUB_REPO}:shipr-frontend
+                    '''
+                }
+            }
+        }
+    }
+
+    post {
+        success {
+            echo 'All services have been successfully deployed locally!'
+        }
+        failure {
+            echo 'Deployment failed. Please check the logs for details.'
+        }
+    }
+}
```

Comment:
Could you please tell me what is this Jenkinsfile doing?

Reply:
This will get the branch name from the user through env variable and then checkout to that branch.

Reply:
@osama-azm  Is there anything the branch is being used for? I do not see any use of getting the branch name here. The requirements stated

- Create CI/CD pipeline on Jenkins
- Developer should go to jenkins and select the branch he or she wants to deploy and code in that branch should be built.

You need to create 5 separate CI pipelines and 5 separate CD pipelines for this task. The CD pipelines will be used to deploy to k8s using helm. Why 5?

1. Shipr Frontend
2. Shipr Payment service
3. Shipr Payment service consumer
4. Shipr Inventory service
5. Shipr Inventory service consumer

Reply:
Created single pipeline that will handle the CI and CD of all the services.

Reply:
Added the separate jenkinsfile for different services that will get the branch name from user. (Tested on different repos for the provided services)